### PR TITLE
chore(ui): Improve sidebar specs

### PR DESCRIPTION
**Pull request by gggritso**
Follow-up to https://github.com/getsentry/sentry/pull/72012. Updates the tests a bit, moves things around, and expands test coverage.

- Move link set tests together

- Simplify `ConfigStore` init
    
    - init in `beforeEach`
    - remove `act`

- Update spec structure
    
    1. Set up all possible sidebar feature flags
    2. Await `Broadcast` mount
    3. Fetch elements by role, not by ID
    4. Check element by accessible title, not by ID

### CODE REVIEW

**Review by hubertdeng123 — APPROVED** [review]
thanks for making this cleaner!

File: static/app/components/sidebar/index.spec.tsx [hunk 1]
```diff
@@ -295,102 +294,152 @@ describe('Sidebar', function () {
...
+    beforeEach(function () {
+      ConfigStore.init();
+      ConfigStore.set('features', new Set([]));
+      ConfigStore.set('user', user);
+    });
+
+    it('renders navigation', function () {
+      renderSidebar({organization});
+
+      waitFor(function () {
```
Comment (scttcper) [inline static/app/components/sidebar/index.spec.tsx:307]: waitFor returns a promise that must be awaited

File: static/app/components/sidebar/index.spec.tsx [hunk 2]
```diff
@@ -295,102 +294,152 @@ describe('Sidebar', function () {
...
+        screen.getByRole('navigation', {name: 'Primary Navigation'})
+      ).toBeInTheDocument();
+    });
+
+    it('in self-hosted-errors-only mode, only shows links to basic features', function () {
+      ConfigStore.set('isSelfHostedErrorsOnly', true);
+
+      renderSidebarWithFeatures(ALL_AVAILABLE_FEATURES);
+
+      waitFor(function () {
```
Comment (scttcper) [inline static/app/components/sidebar/index.spec.tsx:321]: same here

File: static/app/components/sidebar/index.spec.tsx [hunk 3]
```diff
@@ -295,102 +294,152 @@ describe('Sidebar', function () {
...
+        expect(apiMocks.broadcasts).toHaveBeenCalled();
+      });
+
+      const links = screen.getAllByRole('link');
```
Comment (scttcper) [inline static/app/components/sidebar/index.spec.tsx:325]: 👍 much better than the querySelectorAll that was here

**Comment by gggritso** [comment]
@scttcper fixed the `waitFor`s, thanks!

**Pull request merged**